charts/osm/templates/osm-configmap.yaml

```diff
@@ -9,6 +9,7 @@ data:
   permissive_traffic_policy_mode: {{ .Values.OpenServiceMesh.enablePermissiveTrafficPolicy | default "false" | quote }}
   egress: {{ .Values.OpenServiceMesh.enableEgress | quote }}
   envoy_log_level: {{ .Values.OpenServiceMesh.envoyLogLevel | quote }}
+  envoy_image: {{ .Values.OpenServiceMesh.sidecarImage | quote }}
```

Comment:
I think everything adding to osm-configmap.yaml now needs to be sync'ed in MeshConfig CRD and struct definition. 

cc. @shashankram

Reply:
@jaellio Would you mind adding this to the [MeshConfig CRD](https://github.com/openservicemesh/osm/blob/main/charts/osm/crds/meshconfig.yaml) and [client code](https://github.com/openservicemesh/osm/blob/main/pkg/apis/config/v1alpha1/mesh_config.go) in a subsequent change. The autogen client code needs to be regenerated using `./codegen/gen-crd-client.sh`.

Reply:
The changes have been added to this PR